### ISSUE TITLE
Webpack loaders not found when building with `next@13.2.1`

### ISSUE THREAD

**Issue by LuckeeDev** [vercel/next.js#46374]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Home
Binaries:
      Node: 16.19.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
Relevant packages:
      next: 13.2.1
      eslint-config-next: 13.1.1
      react: 18.2.0
      react-dom: 18.2.0

### Link to the code that reproduces this issue

https://github.com/LuckeeDev/csl

### To Reproduce

Install `next@13.2.1` and run `yarn build web`. The error will prevent the build from completing, whereas with `next@13.1.6` everything works fine.

### Describe the Bug

I get multiple statements like the following when building the project:
```
Module parse failed: The keyword 'interface' is reserved (4:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. S
ee https://webpack.js.org/concepts#loaders
| import axios from 'axios';
|
> interface TokensResponse {
|       access_token: string;
|       expires_in: number;
```

### Expected Behavior

The build should work without errors.

**Comment by wescopeland** [comment]
Possibly related https://github.com/vercel/next.js/pull/45455

**Comment by sabbircste** [comment]
Facing Issue for this version : Module parse failed: Unexpected token (6:5) You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file.

**Comment by anonmily** [comment]
Having this issue as well. Tried downgrading to 13.1.16 and was able to build with no errors without changing anything else, so it definitely does seem to be a problem with 13.2...

**Comment by luke-h1** [comment]
Also having this problem with `next@13.2.1` using vanilla extract & the `createVanillaExtractPlugin` in `next.config.js`:

```typescript
./node_modules/.pnpm/@vanilla-extract+webpack-plugin@2.2.0/node_modules/@vanilla-extract/webpack-plugin/extracted.js.webpack[javascript/auto]!=!./node_modules/.pnpm/next@13.2.1_3duxcpitbtplz62feflag7fwby/node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[4].oneOf[0].use[1]!./node_modules/.pnpm/next@13.2.1_3duxcpitbtplz62feflag7fwby/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[4].oneOf[0].use[2]!./node_modules/.pnpm/@vanilla-extract+webpack-plugin@2.2.0/node_modules/@vanilla-extract/webpack-plugin/virtualFileLoader/dist/vanilla-extract-webpack-plugin-virtualFileLoader.cjs.js?{"fileName":"src/components/Footer/ThemeSelect/ThemeSelect.css.ts.vanilla.css","source":"Ll8xOW10cGQwIHsKICBwb3NpdGlvbjogcmVsYXRpdmU7CiAgZGlzcGxheTogaW5saW5lLWZsZXg7CiAgaGVpZ2h0OiAycmVtOwogIHdpZHRoOiBtaW4tY29udGVudDsKfQouXzE5bXRwZDEgewogIHBvc2l0aW9uOiBhYnNvbHV0ZTsKICBkaXNwbGF5OiBncmlkOwogIHBsYWNlLWl0ZW1zOiBjZW50ZXI7CiAgbGVmdDogdmFyKC0temhsN3hnZCk7CiAgdG9wOiAwOwogIHdpZHRoOiB2YXIoLS16aGw3eGdlKTsKICBoZWlnaHQ6IDEwMCU7CiAgcG9pbnRlci1ldmVudHM6IG5vbmU7Cn0KLl8xOW10cGQyIHsKICBib3JkZXI6IDFweCBzb2xpZDsKICBib3JkZXItY29sb3I6IHZhcigtLWNvbG9yLWJvcmRlcik7CiAgcGFkZGluZy10b3A6IHZhcigtLXpobDd4Z2MpOwogIHBhZGRpbmctYm90dG9tOiB2YXIoLS16aGw3eGdjKTsKICBwYWRkaW5nLXJpZ2h0OiBjYWxjKCh2YXIoLS16aGw3eGdkKSAqIDIpICsgdmFyKC0temhsN3hnZSkpOwogIHBhZGRpbmctbGVmdDogY2FsYygodmFyKC0temhsN3hnZCkgKiAyKSArIHZhcigtLXpobDd4Z2UpKTsKICBoZWlnaHQ6IDJyZW07CiAgZm9udC1zaXplOiB2YXIoLS16aGw3eGczKTsKICBib3JkZXItcmFkaXVzOiB2YXIoLS16aGw3eGdtKTsKICB0ZXh0LWFsaWduOiBsZWZ0Owp9Ci5fMTltdHBkMjpmb2N1cyB7CiAgb3V0bGluZTogdHJhbnNwYXJlbnQ7Cn0KLl8xOW10cGQyOmZvY3VzLXZpc2libGUgewogIG91dGxpbmUtd2lkdGg6IDJweDsKICBvdXRsaW5lLXN0eWxlOiBzb2xpZDsKICBvdXRsaW5lLW9mZnNldDogMnB4OwogIG91dGxpbmUtY29sb3I6IHZhcigtLWNvbG9yLW91dGxpbmUpOwp9Ci5fMTltdHBkMyB7CiAgcG9zaXRpb246IGFic29sdXRlOwogIGRpc3BsYXk6IGdyaWQ7CiAgcGxhY2UtaXRlbXM6IGNlbnRlcjsKICByaWdodDogdmFyKC0temhsN3hnZCk7CiAgdG9wOiAwOwogIHdpZHRoOiB2YXIoLS16aGw3eGdlKTsKICBoZWlnaHQ6IDEwMCU7CiAgcG9pbnRlci1ldmVudHM6IG5vbmU7Cn0="}!./node_modules/.pnpm/@vanilla-extract+webpack-plugin@2.2.0/node_modules/@vanilla-extract/webpack-plugin/extracted.js
TypeError: options.postcss is not a function
    at /Users/lukehive/srv/dev/lhowsam.com/node_modules/.pnpm/next@13.2.1_3duxcpitbtplz62feflag7fwby/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js:37:55
    at Span.traceAsyncFn (/Users/lukehive/srv/dev/lhowsam.com/node_modules/.pnpm/next@13.2.1_3duxcpitbtplz62feflag7fwby/node_modules/next/dist/trace/trace.js:79:26)
    at Object.loader (/Users/lukehive/srv/dev/lhowsam.com/node_modules/.pnpm/next@13.2.1_3duxcpitbtplz62feflag7fwby/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js:12:16)

Import trace for requested module:
./node_modules/.pnpm/@vanilla-extract+webpack-plugin@2.2.0/node_modules/@vanilla-extract/webpack-plugin/extracted.js.webpack[javascript/auto]!=!./node_modules/.pnpm/next@13.2.1_3duxcpitbtplz62feflag7fwby/node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[4].oneOf[0].use[1]!./node_modules/.pnpm/next@13.2.1_3duxcpitbtplz62feflag7fwby/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[4].oneOf[0].use[2]!./node_modules/.pnpm/@vanilla-extract+webpack-plugin@2.2.0/node_modules/@vanilla-extract/webpack-plugin/virtualFileLoader/dist/vanilla-extract-webpack-plugin-virtualFileLoader.cjs.js?{"fileName":"src/components/Footer/ThemeSelect/ThemeSelect.css.ts.vanilla.css","source":"Ll8xOW10cGQwIHsKICBwb3NpdGlvbjogcmVsYXRpdmU7CiAgZGlzcGxheTogaW5saW5lLWZsZXg7CiAgaGVpZ2h0OiAycmVtOwogIHdpZHRoOiBtaW4tY29udGVudDsKfQouXzE5bXRwZDEgewogIHBvc2l0aW9uOiBhYnNvbHV0ZTsKICBkaXNwbGF5OiBncmlkOwogIHBsYWNlLWl0ZW1zOiBjZW50ZXI7CiAgbGVmdDogdmFyKC0temhsN3hnZCk7CiAgdG9wOiAwOwogIHdpZHRoOiB2YXIoLS16aGw3eGdlKTsKICBoZWlnaHQ6IDEwMCU7CiAgcG9pbnRlci1ldmVudHM6IG5vbmU7Cn0KLl8xOW10cGQyIHsKICBib3JkZXI6IDFweCBzb2xpZDsKICBib3JkZXItY29sb3I6IHZhcigtLWNvbG9yLWJvcmRlcik7CiAgcGFkZGluZy10b3A6IHZhcigtLXpobDd4Z2MpOwogIHBhZGRpbmctYm90dG9tOiB2YXIoLS16aGw3eGdjKTsKICBwYWRkaW5nLXJpZ2h0OiBjYWxjKCh2YXIoLS16aGw3eGdkKSAqIDIpICsgdmFyKC0temhsN3hnZSkpOwogIHBhZGRpbmctbGVmdDogY2FsYygodmFyKC0temhsN3hnZCkgKiAyKSArIHZhcigtLXpobDd4Z2UpKTsKICBoZWlnaHQ6IDJyZW07CiAgZm9udC1zaXplOiB2YXIoLS16aGw3eGczKTsKICBib3JkZXItcmFkaXVzOiB2YXIoLS16aGw3eGdtKTsKICB0ZXh0LWFsaWduOiBsZWZ0Owp9Ci5fMTltdHBkMjpmb2N1cyB7CiAgb3V0bGluZTogdHJhbnNwYXJlbnQ7Cn0KLl8xOW10cGQyOmZvY3VzLXZpc2libGUgewogIG91dGxpbmUtd2lkdGg6IDJweDsKICBvdXRsaW5lLXN0eWxlOiBzb2xpZDsKICBvdXRsaW5lLW9mZnNldDogMnB4OwogIG91dGxpbmUtY29sb3I6IHZhcigtLWNvbG9yLW91dGxpbmUpOwp9Ci5fMTltdHBkMyB7CiAgcG9zaXRpb246IGFic29sdXRlOwogIGRpc3BsYXk6IGdyaWQ7CiAgcGxhY2UtaXRlbXM6IGNlbnRlcjsKICByaWdodDogdmFyKC0temhsN3hnZCk7CiAgdG9wOiAwOwogIHdpZHRoOiB2YXIoLS16aGw3eGdlKTsKICBoZWlnaHQ6IDEwMCU7CiAgcG9pbnRlci1ldmVudHM6IG5vbmU7Cn0="}!./node_modules/.pnpm/@vanilla-extract+webpack-plugin@2.2.0/node_modules/@vanilla-extract/webpack-plugin/extracted.js
src/components/Footer/ThemeSelect/ThemeSelect.css.ts.vanilla.css!=!./node_modules/.pnpm/@vanilla-extract+webpack-plugin@2.2.0/node_modules/@vanilla-extract/webpack-plugin/virtualFileLoader/dist/vanilla-extract-webpack-plugin-virtualFileLoader.cjs.js?{"fileName":"src/components/Footer/ThemeSelect/ThemeSelect.css.ts.vanilla.css","source":"Ll8xOW10cGQwIHsKICBwb3NpdGlvbjogcmVsYXRpdmU7CiAgZGlzcGxheTogaW5saW5lLWZsZXg7CiAgaGVpZ2h0OiAycmVtOwogIHdpZHRoOiBtaW4tY29udGVudDsKfQouXzE5bXRwZDEgewogIHBvc2l0aW9uOiBhYnNvbHV0ZTsKICBkaXNwbGF5OiBncmlkOwogIHBsYWNlLWl0ZW1zOiBjZW50ZXI7CiAgbGVmdDogdmFyKC0temhsN3hnZCk7CiAgdG9wOiAwOwogIHdpZHRoOiB2YXIoLS16aGw3eGdlKTsKICBoZWlnaHQ6IDEwMCU7CiAgcG9pbnRlci1ldmVudHM6IG5vbmU7Cn0KLl8xOW10cGQyIHsKICBib3JkZXI6IDFweCBzb2xpZDsKICBib3JkZXItY29sb3I6IHZhcigtLWNvbG9yLWJvcmRlcik7CiAgcGFkZGluZy10b3A6IHZhcigtLXpobDd4Z2MpOwogIHBhZGRpbmctYm90dG9tOiB2YXIoLS16aGw3eGdjKTsKICBwYWRkaW5nLXJpZ2h0OiBjYWxjKCh2YXIoLS16aGw3eGdkKSAqIDIpICsgdmFyKC0temhsN3hnZSkpOwogIHBhZGRpbmctbGVmdDogY2FsYygodmFyKC0temhsN3hnZCkgKiAyKSArIHZhcigtLXpobDd4Z2UpKTsKICBoZWlnaHQ6IDJyZW07CiAgZm9udC1zaXplOiB2YXIoLS16aGw3eGczKTsKICBib3JkZXItcmFkaXVzOiB2YXIoLS16aGw3eGdtKTsKICB0ZXh0LWFsaWduOiBsZWZ0Owp9Ci5fMTltdHBkMjpmb2N1cyB7CiAgb3V0bGluZTogdHJhbnNwYXJlbnQ7Cn0KLl8xOW10cGQyOmZvY3VzLXZpc2libGUgewogIG91dGxpbmUtd2lkdGg6IDJweDsKICBvdXRsaW5lLXN0eWxlOiBzb2xpZDsKICBvdXRsaW5lLW9mZnNldDogMnB4OwogIG91dGxpbmUtY29sb3I6IHZhcigtLWNvbG9yLW91dGxpbmUpOwp9Ci5fMTltdHBkMyB7CiAgcG9zaXRpb246IGFic29sdXRlOwogIGRpc3BsYXk6IGdyaWQ7CiAgcGxhY2UtaXRlbXM6IGNlbnRlcjsKICByaWdodDogdmFyKC0temhsN3hnZCk7CiAgdG9wOiAwOwogIHdpZHRoOiB2YXIoLS16aGw3eGdlKTsKICBoZWlnaHQ6IDEwMCU7CiAgcG9pbnRlci1ldmVudHM6IG5vbmU7Cn0="}!./node_modules/.pnpm/@vanilla-extract+webpack-plugin@2.2.0/node_modules/@vanilla-extract/webpack-plugin/extracted.js
./src/components/Footer/ThemeSelect/ThemeSelect.css.ts
./src/components/Footer/ThemeSelect/ThemeSelect.tsx
./src/components/Footer/Footer.tsx

```

**Comment by timneutkens** [comment]
Seems NX uses Next.js internals that are not documented and do not fall under any guarantee that they'll exist in the future given that they're not documented and not part of the public api. e.g. importing `next/dist/x` is a clear marker that internals are being used.
These internals have now changed in 13.2 to allow running the webpack build in a separate worker to further parallelize the build steps across multiple cores, which will be enabled soon for all builds.

**Comment by shuding** [comment]
#46490 should fix the vanilla-extract problem. There's also something to add in vanilla-extract side and I'll comment in https://github.com/vanilla-extract-css/vanilla-extract/issues/1013

**Comment by karlhorky** [comment]
Is this also the reason that [this StackBlitz reproduction](https://stackblitz.com/edit/vercel-next-js-tteixg?file=app%2Fpage.tsx) errors out when using `bcrypt` in a Route Handler? 

Error message:

```
error - ./node_modules/@mapbox/node-pre-gyp/lib/util/nw-pre-gyp/index.html
Module parse failed: Unexpected token (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
> <!doctype html>
| <html>
| <head>

Import trace for requested module:
./node_modules/@mapbox/node-pre-gyp/lib/util/nw-pre-gyp/index.html
./node_modules/@mapbox/node-pre-gyp/lib/ sync ^\.\/.*$
./node_modules/@mapbox/node-pre-gyp/lib/node-pre-gyp.js
./node_modules/bcrypt/bcrypt.js
./app/api/route.ts
```

**Comment by sabbircste** [comment]
Is this solved?

**Comment by skaneprime** [comment]
> Is this solved?

Not yet. Having trouble to build next 13.2.x in NX 15.7.2

**Comment by erikmellum** [comment]
We had to downgrade nextjs for product deployments to 13.1.6 as others have suggested. font-loader-manifest.json was creating an error for us. Sharing for others who search for this issue since it wasn't easy for us to locate and cost us some dev time.

<img width="897" alt="image" src="https://user-images.githubusercontent.com/6475033/222220967-4f28415c-fe7e-41b5-9ab2-93dbb87e1538.png">

**Comment by rexwangcc** [comment]
Are there any updates on this? We faced the same issue on `13.2.3` and the error trace is still the same as above

```
Error: Cannot find module '/FOO/.next/server/font-loader-manifest.json'
Require stack:
- /FOO/node_modules/next/dist/server/next-server.js
- /FOO/server.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at mod._resolveFilename (/FOO/node_modules/next/dist/build/webpack/require-hook.js:23:32)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at NextNodeServer.getFontLoaderManifest (/FOO/node_modules/next/dist/server/next-server.js:643:16)
    at new Server (/FOO/node_modules/next/dist/server/base-server.js:74:40)
    at new NextNodeServer (/FOO/node_modules/next/dist/server/next-server.js:69:9)
    at Server.<anonymous> (/FOO/server.js:34:22)
    at Object.onceWrapper (node:events:627:28) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/FOO/node_modules/next/dist/server/next-server.js',
    '/FOO/server.js'
  ]
}

```

**Comment by lucasmerlin** [comment]
```bash
nx migrate 15.9.0-beta.2
```
fixes it for me

https://github.com/nrwl/nx/releases/tag/15.9.0-beta.2 includes https://github.com/nrwl/nx/pull/15650 which should fix this 🎉

**Comment by ryandegruyter** [comment]
Tried the upgrade to 15.9.0-beta.2 but it results in different errors:

```
Attempted import error: 'ReactComponent' is not exported from '../../public/img/logos/alpha.svg' (imported as 'AlphaSvg').
```

And

```
info  - Generating static pages (0/45)
Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at Zc (/Users/.../node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:74:493)
```

**Comment by LuckeeDev** [comment]
I'm closing this issue since it was fixed in [`nx@15.8.7`](https://github.com/nrwl/nx/releases/tag/15.8.7) thank to this pull request: [#15650](https://github.com/nrwl/nx/pull/15650).
